### ISSUE TITLE
allow configuring how manually curried functions are formatted

### ISSUE THREAD

**Issue by adrian-gierakowski** [dprint/dprint-plugin-typescript#388]
First I'd like to thank you for creating this project. It's really nice to have an alternative to prettier which is not only faster but also more respectful in regards to developers formatting choices.

However there is one issue which is preventing me from introducing dprint at work. In our code base, we've got a lot of functions which look like this:

```js
const curriedFunc = a => b => c => {
  return a + b + c
}
```

dprint formats then as follows:

```js
const curriedFunc = a =>
  b =>
    c => {
      return a + b + c
    }
```

I'd like the function to remain unchanged, unless the first line exceeds the line char limit, in which case I'd like it to be formatted like this:

```js
const curriedFunc = 
  a => 
  b => 
  c => {
    return a + b + c
  }
```  

Note that the following function currently is left as is by the formatter, but the actual functions in my code have bodies which do not fit on a single line

```js
const curriedFunc = a => b => c => a + b + c
``` 

I experimented with different settings but could not achieve the desired result. Do you think it would be possible to make dprint behave as described above? Thanks!

**Comment by adrian-gierakowski** [comment]
@dsherret anything I could do to help with addressing this? If you point out which part of the code base one I’d have to look into to change this behaviour, I might be able to commit some time to work on this. Thanks!

**Comment by dsherret** [comment]
It's complicated and hard. I have this half done and will try to get it merged in the next couple of days. I just forgot about this.

**Comment by adrian-gierakowski** [comment]
Oh wow, thanks for looking into it so quickly!

**Comment by dsherret** [comment]
@adrian-gierakowski should be fixed in 0.71.0. Let me know if you run into any problems because I usually don't code with functions like that.

**Comment by adrian-gierakowski** [comment]
thanks a lot @dsherret, this works great!

There is one minor thing that could be improved: currently, if function arguments don't fit on a single line, they are all dropped one line below the `=` sign, and if after that they fit within `lineWidth`, they are all left on that one line. For example. with `lineWidth = 40`, we have:

before:
```js
const func = arg1 => arg2 => arg3 => arg4 => console.log('hi')
```

after:
```js
const func =
  arg1 => arg2 => arg3 => arg4 =>
    console.log('hi')
```

whereas I'd prefer to end up with:

```js
const func =
  arg1 =>
  arg2 =>
  arg3 =>
  arg4 => console.log('hi')
```

similarly to how non-curried functions are formatted:

before:
```js
const func = (arg1, arg2, arg3, arg4, arg5) => console.log('hi')
```

after:
```js
const func = (
  arg1,
  arg2,
  arg3,
  arg4,
  arg5
) => console.log('hi')
```

moreover, with non-curried functions, dprint formats arguments across multiple lines if there is at least one line break already present, while curried functions get collapsed to a single line if they can fit:

before:
```js
const func1 = (
  arg1, arg2) => 1

const func2 =
  arg1 =>
  arg2 => arg3 => 1


const func3 =
  arg1 =>
  arg2 => arg3 => arg4 => 1
```

after 
```js
const func1 = (
  arg1,
  arg2
) => 1

const func2 = arg1 => arg2 => arg3 => 1

const func3 =
  arg1 => arg2 => arg3 => arg4 => 1
```

I'd prefer `func2\3` to be formatted like:

```js
const func2 =
  arg1 => 
  arg2 =>
  arg3 => 
  arg4 => 1
```

For reference, here's the config I used for the examples above:
```json
{
  "typescript": {
    "lineWidth": 40,
    "semiColons": "asi",
    "quoteStyle": "preferSingle",
    "quoteProps": "asNeeded",
    "nextControlFlowPosition": "sameLine",
    "trailingCommas": "never",
    "arrowFunction.useParentheses": "preferNone",
    "binaryExpression.linePerExpression": true,
    "jsx.quoteStyle": "preferSingle",
    "memberExpression.linePerExpression": true,
    "enumDeclaration.memberSpacing": "newLine"
  },
  "json": {},
  "markdown": {},
  "includes": [
    "**/*.{ts,tsx,js,jsx,cjs,mjs,json,md}"
  ],
  "excludes": [
    "**/node_modules",
    "**/*-lock.json"
  ],
  "plugins": [
    "https://plugins.dprint.dev/typescript-0.71.1.wasm",
  ]
}
```

These are not deal breakers though, and what you've done so far allowed me to finally use dprint at work, so thanks again!
